src/frontend/src/icp/components/send/IcSendAmount.svelte

```diff
@@ -50,62 +43,42 @@
 				tokenDecimals: $tokenDecimals,
 				i18n: $i18n
 			});
-
-			if (nonNullish(amountError)) {
-				return;
+		} else if (isNetworkIdEthereum(networkId)) {
+			if ($tokenCkEthLedger) {
+				amountError = assertCkETHMinWithdrawalAmount({
+					amount: value,
+					tokenDecimals: $tokenDecimals,
+					tokenSymbol: $tokenSymbol,
+					minterInfo: $ckEthMinterInfoStore?.[$ckEthereumNativeTokenId],
+					i18n: $i18n
+				});
+			} else {
+				amountError = assertCkETHMinFee({
+					amount: value,
+					tokenSymbol: $tokenSymbol,
+					fee,
+					i18n: $i18n
+				});
 			}
-		}
-
-		if (isNetworkIdEthereum(networkId) && $tokenCkEthLedger) {
-			amountError = assertCkETHMinWithdrawalAmount({
-				amount: value,
-				tokenDecimals: $tokenDecimals,
-				tokenSymbol: $tokenSymbol,
-				minterInfo: $ckEthMinterInfoStore?.[$ckEthereumNativeTokenId],
-				i18n: $i18n
-			});
-
-			if (nonNullish(amountError)) {
-				return;
+		} else {
+			const total = value.add(fee);
+			if (total.gt($balance ?? BigNumber.from(0n))) {
+				amountError = new IcAmountAssertionError($i18n.send.assertion.insufficient_funds);
 			}
 		}
 
-		if (isNetworkIdEthereum(networkId)) {
-			amountError = assertCkETHMinFee({
-				amount: value,
-				tokenSymbol: $tokenSymbol,
-				fee,
-				i18n: $i18n
-			});
-			return;
-		}
-
-		const total = value.add(fee);
-
-		if (total.gt($balance ?? BigNumber.from(0n))) {
-			amountError = new IcAmountAssertionError($i18n.send.assertion.insufficient_funds);
-			return;
-		}
-
-		amountError = undefined;
+		return amountError;
 	};
 
-	const debounceValidate = debounce(validate);
+	const customValidations = [validate];
 
-	$: amount, fee, $ckBtcMinterInfoStore, $ckEthMinterInfoStore, debounceValidate();
+	let reactivityVariables: (unknown | undefined)[];
```

Comment:
have you thought about exposing the function `debounceValidate` to the parent? don't like the pattern?

```
$: fee, $ckBtcMinterInfoStore, $ckEthMinterInfoStore, input.debounceValidate()
```

if you rather like to use the array, can we maybe find a less generic name for the property?

Reply:
Reverted